lib/usb/usb_standard.c

```diff
@@ -277,13 +203,22 @@ usb_standard_set_configuration(usbd_device *usbd_dev,
 	usbd_dev->current_config = found_index + 1;
 
 	if (usbd_dev->current_config > 0) {
-		cfg = &usbd_dev->config[usbd_dev->current_config - 1];
+		cfg = usbd_dev->config[usbd_dev->current_config - 1];
 
 		/* reset all alternate settings configuration */
-		for (i = 0; i < cfg->bNumInterfaces; i++) {
-			if (cfg->interface[i].cur_altsetting) {
-				*cfg->interface[i].cur_altsetting = 0;
+		int totalLen = cfg->wTotalLength;
```

Comment:
nit: `uint16_t totalLen = `... and drop `(int)` cast below? (it is a bit more brittle though)

Reply:
nah